### PR TITLE
CORE-8020: Replacement of `RPCOps` to `RestResource` and `PluggableRestResource`

### PR DESCRIPTION
**Pull request by pmck91**
This PR is for the replacement of `RPCOps` to `RestResource` and `PluggableRestResource`.

There will be future reviews for file names, variable names, and finally the module name.

### CODE REVIEW

File: libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/PluggableRestResource.kt [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package net.corda.httprpc
```
Comment (vkolomeyko) [inline libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/PluggableRestResource.kt:1]: Given that no module move been performed as part of this PR, I think title/description of the PR needs to be changed.

File: components/configuration/configuration-rpcops-service-impl/src/test/kotlin/net/corda/configuration/rpcops/impl/ConfigRPCOpsImplTests.kt [hunk 2]
```diff
@@ -263,15 +263,15 @@ class ConfigRPCOpsImplTests {
         )).isEqualTo(configRPCOps.get(configSection))
     }
 
-    /** Returns a [ConfigRPCOps] where the RPC sender returns [future] in response to any RPC requests.
+    /** Returns a [ConfigRestResource] where the RPC sender returns [future] in response to any RPC requests.
```
Comment (ben-millar) [inline components/configuration/configuration-rpcops-service-impl/src/test/kotlin/net/corda/configuration/rpcops/impl/ConfigRPCOpsImplTests.kt:266]: Comment still mentions RPC

File: libs/http-rpc/http-rpc-client/src/main/kotlin/net/corda/httprpc/client/HttpRpcClient.kt [hunk 3]
```diff
@@ -18,16 +18,18 @@ import java.util.concurrent.CopyOnWriteArraySet
...
  * @property clientConfig The configuration for the client to use.
- * @property healthCheckInterval The interval on which health check calls to the server will happen, ensuring connectivity.
+ * @property healthCheckInterval The interval on which health check calls to the server will happen, ensuring
+ *                               connectivity.
  */
 @Suppress("LongParameterList")
-class HttpRpcClient<I : RpcOps> internal constructor(
+class HttpRpcClient<I : RestResource> internal constructor(
     private val baseAddress: String,
     private val rpcOpsClass: Class<I>,
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-client/src/main/kotlin/net/corda/httprpc/client/HttpRpcClient.kt:34]: Should this be `RestResourceClass`?

File: libs/http-rpc/http-rpc-client/src/test/kotlin/net/corda/httprpc/client/HttpRpcClientTest.kt [hunk 4]
```diff
@@ -21,15 +21,17 @@ import kotlin.test.assertEquals
 import kotlin.test.assertNotNull
 
 class HttpRpcClientTest {
-    private inline fun <reified I : RpcOps> mockProxyGenerator(rpcOpsClass: Class<I>, proxyHandler: HttpRpcClientProxyHandler<I>): I {
+    private inline fun <reified I : RestResource> mockProxyGenerator(
+        rpcOpsClass: Class<I>, proxyHandler: HttpRpcClientProxyHandler<I>): I {
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-client/src/test/kotlin/net/corda/httprpc/client/HttpRpcClientTest.kt:25]: As above, just checking we still want to be referencing RPC here.

File: libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/HttpRpcServerImpl.kt [hunk 5]
```diff
@@ -80,19 +80,19 @@ class HttpRpcServerImpl(
         }
     }
 
-    private fun getResources(rpcOpsImpls: List<PluggableRPCOps<out RpcOps>>): List<Resource> {
-        log.debug { "Get resources for RPCOps implementations of ${rpcOpsImpls.joinToString()}." }
+    private fun getResources(restResourceImpls: List<PluggableRestResource<out RestResource>>): List<Resource> {
+        log.debug { "Get resources for RPCOps implementations of ${restResourceImpls.joinToString()}." }
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/HttpRpcServerImpl.kt:84]: Do the strings in this file need to be changed to replace mentions of rpc/RPCOps?

File: libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/factory/HttpRpcServerFactoryImpl.kt [hunk 6]
```diff
@@ -16,13 +16,13 @@ import java.util.function.Supplier
 class HttpRpcServerFactoryImpl : HttpRpcServerFactory {
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/factory/HttpRpcServerFactoryImpl.kt:16]: Several mentions of RPC in this class, flagging to be sure to be sure!

File: libs/http-rpc/http-rpc-server-impl/src/test/java/net/corda/httprpc/server/apigen/test/TestJavaPrimitivesRPCopsImpl.java [hunk 7]
```diff
@@ -1,9 +1,9 @@
 package net.corda.httprpc.server.apigen.test;
 
-import net.corda.httprpc.PluggableRPCOps;
+import net.corda.httprpc.PluggableRestResource;
 import org.jetbrains.annotations.NotNull;
 
-public class TestJavaPrimitivesRPCopsImpl implements TestJavaPrimitivesRpcOps, PluggableRPCOps<TestJavaPrimitivesRpcOps> {
+public class TestJavaPrimitivesRPCopsImpl implements TestJavaPrimitivesRestResource, PluggableRestResource<TestJavaPrimitivesRestResource> {
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-server-impl/src/test/java/net/corda/httprpc/server/apigen/test/TestJavaPrimitivesRPCopsImpl.java:6]: Implementing RestResources but class is still named with `RPCops`

File: libs/http-rpc/http-rpc-server-impl/src/test/kotlin/net/corda/httprpc/server/impl/security/HttpRpcSecurityManagerTest.kt [hunk 8]
```diff
@@ -40,7 +40,7 @@ class HttpRpcSecurityManagerTest {
 
         private fun methodFullName(clazz: Class<*>, methodName: String): String {
             require(clazz.isInterface) { "Must be an interface: $clazz" }
-            require(RpcOps::class.java.isAssignableFrom(clazz)) { "Must be assignable from RPCOps: $clazz" }
+            require(RestResource::class.java.isAssignableFrom(clazz)) { "Must be assignable from RPCOps: $clazz" }
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-server-impl/src/test/kotlin/net/corda/httprpc/server/impl/security/HttpRpcSecurityManagerTest.kt:43]: String here should probably be changed to RestResource

File: libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/NumberSequencesRestResource.kt [hunk 9]
```diff
@@ -10,11 +10,11 @@ enum class NumberTypeEnum {
 }
 
 @HttpRpcResource(
-    name = "net.corda.httprpc.server.impl.rpcops.NumberSequencesRPCOps",
+    name = "net.corda.httprpc.server.impl.rpcops.NumberSequencesRestResource",
     description = "Number Sequences RPC Ops",
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/NumberSequencesRestResource.kt:14]: Description may need updating

File: libs/http-rpc/http-rpc-tools/src/main/kotlin/net/corda/httprpc/tools/annotations/validation/EndpointAnnotationValidator.kt [hunk 10]
```diff
@@ -1,13 +1,13 @@
 package net.corda.httprpc.tools.annotations.validation
 
-import net.corda.httprpc.RpcOps
+import net.corda.httprpc.RestResource
 import net.corda.httprpc.annotations.isRpcEndpointAnnotation
 import java.lang.reflect.Method
 
 /**
  * Validates that every method has no more than one HTTP RPC endpoint annotation.
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-tools/src/main/kotlin/net/corda/httprpc/tools/annotations/validation/EndpointAnnotationValidator.kt:8]: Several mentions of RPC here

File: libs/http-rpc/http-rpc-tools/src/main/kotlin/net/corda/httprpc/tools/annotations/validation/HttpRpcInterfaceValidator.kt [hunk 11]
```diff
@@ -10,11 +10,11 @@ object HttpRpcInterfaceValidator {
     /**
      * Validates an interface
      *
-     * @param rpcOpsInterface An interface class extending [RpcOps]
+     * @param rpcOpsInterface An interface class extending [RestResource]
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-tools/src/main/kotlin/net/corda/httprpc/tools/annotations/validation/HttpRpcInterfaceValidator.kt:13]: Should probably be `restResourceInterface`

File: libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/annotations/RPCSinceVersion.kt [hunk 12]
```diff
@@ -4,7 +4,7 @@ package net.corda.httprpc.annotations
  * Records the protocol version in which this RPC method was added.
  *
  * In particular [net.corda.httprpc.client.HttpRpcClient] makes use of this annotation to guard against scenario when
- * HTTP RPC Client's [net.corda.httprpc.RpcOps] interface been updated ahead of server side implementation.
+ * HTTP RPC Client's [net.corda.httprpc.RestResource] interface been updated ahead of server side implementation.
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/annotations/RPCSinceVersion.kt:7]: Mentions RPC

File: processors/rpc-processor/src/integrationTest/kotlin/net/corda/processors/rpc/OpenApiCompatibilityTest.kt [hunk 13]
```diff
@@ -44,37 +44,37 @@ class OpenApiCompatibilityTest {
...
     @Test
     fun test() {
-        val allOps = dynamicRpcOps.map { (it as PluggableRPCOps<*>).targetInterface }.sortedBy { it.name }
+        val allOps = dynamicRpcOps.map { (it as PluggableRestResource<*>).targetInterface }.sortedBy { it.name }
         assertThat(allOps.filterNot {
-            it.name.contains("HelloRpcOps") // the only test, i.e. not important RPC Ops we have
+            it.name.contains("HelloRestResource") // the only test, i.e. not important RPC Ops we have
         }.toSet()).isEqualTo(importantRpcOps)
 
         logger.info("RPC Ops discovered: ${allOps.map { it.simpleName }}")
```
Comment (ben-millar) [inline processors/rpc-processor/src/integrationTest/kotlin/net/corda/processors/rpc/OpenApiCompatibilityTest.kt:80]: 'RPC Ops' -> REST Resources

**Review by ben-millar — CHANGES_REQUESTED** [review]
I've highlighted some instances where RPC verbiage is still used where I'm not sure it should be; please address as appropriate (Maybe they're intentional and will be tackled in a future PR for example). I'm happy to approve once comments are resolved!

File: libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/PluggableRestResource.kt (hunk 1 above)
Comment (pmck91) [reply libs/http-rpc/http-rpc/src/main/kotlin/net/corda/httprpc/PluggableRestResource.kt:1]: Added a description!

File: components/configuration/configuration-rpcops-service-impl/src/test/kotlin/net/corda/configuration/rpcops/impl/ConfigRPCOpsImplTests.kt (hunk 2 above)
Comment (pmck91) [reply components/configuration/configuration-rpcops-service-impl/src/test/kotlin/net/corda/configuration/rpcops/impl/ConfigRPCOpsImplTests.kt:266]: will get in future PR for comments and variable names

**Comment by pmck91** [comment]
> I've highlighted some instances where RPC verbiage is still used where I'm not sure it should be; please address as appropriate (Maybe they're intentional and will be tackled in a future PR for example). I'm happy to approve once comments are resolved!

good catches, I will address all of the variable and class names as well as comments in my next pr to avoid this one becoming even larger

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by ben-millar — APPROVED** [review]

**Pull request merged**